libethash-cuda/CUDAMiner.h

```diff
@@ -139,6 +141,8 @@ class CUDAMiner: public Miner
 
 	static unsigned s_numInstances;
 	static int s_devices[16];
+
+	static bool s_noeval;
```

Comment:
Actually, can we stop making these flags global? What the point?

Reply:
We could. My understanding is that they are static members shared by all instances of the class. Isn't that the idea here?

Reply:
Certainly that's the case for s_devices and s_numInstances

Reply:
Thinking this through, perhaps not! There is a single instance of the CUDAMiner class.

Ok. I'm convinced. Let me make these changes.

Reply:
I have not looked into details there, but maintaining these static flags seem to be more trouble. I think we have an instance per GPU, but having a set of int in each does not seem to be a big deal.
I'm not sure how much work it is to change this. If it is not very simple, we can postpone it to another PR.

In future we might want to have an individual options per GPU, so this should go in this direction.

Reply:
Yes, I ran into exactly such a case recently where different parameters per Radeon card type would be more optimal. Not sure how the command line syntax would be expanded to handle per card parameters.

Reply:
I'd prefer to move to per card now (just the variables) if you don't mind? I'm afraid that putting it off it will never get done. Not planning any immediate per card specific functionality though.

Reply:
Do you plan to do more changes then?

Reply:
Yes, look at these one by one and move to private m_ type where applicable.

Reply:
@chfast Nevermind... Not so simple! These are all static variables set in Mineraux using static functions before the actual per card instances are instantiated. I won't be doing more in this one.

Reply:
@chfast BTW. I can't really find exactly where the miners get instantiated. Don't get this 'sealer' stuff either?

Reply:
Sealer is another unnecessary abstraction...